subsys/bluetooth/services/fast_pair/fp_keys.c

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <device.h>
+#include <sys/byteorder.h>
+#include <bluetooth/conn.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(fp_keys, CONFIG_BT_FAST_PAIR_LOG_LEVEL);
+
+#include "fp_registration_data.h"
+#include "fp_keys.h"
+#include "fp_crypto.h"
+#include "fp_common.h"
+#include "fp_storage.h"
+
+#define EMPTY_AES_KEY_BYTE	0xff
+#define FP_ACCOUNT_KEY_PREFIX	0x04
+
+#define FP_KEY_TIMEOUT				K_SECONDS(10)
+#define FP_KEY_GEN_FAILURE_MAX_CNT		10
+#define FP_KEY_GEN_FAILURE_CNT_RESET_TIMEOUT	K_MINUTES(5)
+
+enum fp_state {
+	FP_STATE_INITIAL,
+	FP_STATE_USE_TEMP_KEY,
+	FP_STATE_WAIT_FOR_ACCOUNT_KEY,
+};
+
+struct fp_procedure {
+	enum fp_state state;
+	uint8_t key_gen_failure_cnt;
+	bool pairing_mode;
+	uint8_t aes_key[FP_ACCOUNT_KEY_LEN];
+	struct k_work_delayable timeout;
```

Comment:
Moved. Still I think it would not change anything, because the padding would still be added at the end of the struct to align size of the whole structure.